### PR TITLE
[REVIEW] Add doc note on conda channel_priority

### PR DESCRIPTION
**Pull request by rongou**
Add a note that `channel_priority` should be disabled when creating conda environment from the yml file. Otherwise will get package conflicts.

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by kkraus14** [comment]
@rongou can you share your `.condarc`? In general strict channel priority should help us, but I suspect that an old cupy in the rapidsai or rapidsai-nightly channels could mess with things.

**Comment by rongou** [comment]
This is the version that doesn't work:
```
auto_activate_base: false
channels:
  - conda-forge
  - defaults
channel_priority: strict
```
Once I changed it to `channel_priority: disabled`, it started working.

**Comment by kkraus14** [comment]
@jakirkham any ideas on what the behavior / expected behavior of the conda solver is when using an environment with specified channels plus a `.condarc` with `channel_priority: strict`? Unsure what channels it would prioritize.

**Comment by jakirkham** [comment]
I wouldn't disable `channel_priority`. That's likely to cause problems.

However the way we are doing RC packages doesn't mesh well with `strict`. Would either leave it unspecified (using default) or specify `flexible`, which is the default.

Long term it would be good to figure out how to get `strict` to work for us. It should really help solve some problems with bad solves.

**Comment by rongou** [comment]
Yes verified setting `channel_priority` to `flexible` also works.

I think the problem with `strict` is that some packages need to come from lower priority channels. If we know what these are, we can specify the channel for each package in the yml file. https://stackoverflow.com/questions/58555389/set-the-channel-priority-in-conda-environment-yaml

**Comment by jakirkham** [comment]
Thanks for checking. 🙂

It sounds like that is more a question of how to encode this preference in environment files, which seems like a different (albeit important) discussion.

For reference here is the discussion about using strict with an RC channel ( https://github.com/conda/conda/issues/8752 ). Some of that happened in a video conference, but we did try to summarize the conclusion in that thread.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -157,6 +157,7 @@ git submodule update --init --remote --recursive
 - Create the conda development environment `cudf_dev`:
 ```bash
 # create the conda environment (assuming in base `cudf` directory)
+# note: `channel_priority` cannot be set to `strict`; use `flexible` instead
```
Comment (jakirkham) [inline CONTRIBUTING.md]: What do you think about a rewording like this? 🙂

```suggestion
# note: RAPIDS currently doesn't support `channel_priority: strict`; use `channel_priority: flexible` instead
```

Comment (rongou) [reply CONTRIBUTING.md]: Done.

**Comment by kkraus14** [comment]
@rongou could you add a CHANGELOG entry? This is then ready to be merged. Thanks!

**Comment by rongou** [comment]
@kkraus14 done.

**Comment by jakirkham** [comment]
LGTM. Thanks @rongou! 😀

**Pull request merged**